### ISSUE TITLE
get rid of hex conversion

### ISSUE THREAD

**Issue by bgrosse-midokura** [midonet/tssrp6a#21]
in `utils.ts`, `bigIntegerToWordArray` and `wordArrayToBigInteger` should not use an intermediate "hex" format.

**Comment by bufistov** [comment]
I tried to do it, but BigInteger internal representation blowed up my mind :)
BigInteger has toByteArray() function that we can use to avoid string building in bigIntegerToWord() conversion. toByteArray() cannot be used directly if zero byte of positive number has most significant bit set, in this case the first byte is "sign" byte and should be ignored. The overall code is something like this:
```JavaScript
export const bigIntegerToWordArray = (n: BigInteger): HashWordArray => {
  const bytes = n.toByteArray();
  const firstByte = (bytes.length > 1 && bytes[0] === 0) ? 1 : 0;
  const numBytes = bytes.length - firstByte;
  const wordArray: number[] = new Array(ceilDiv4(numBytes));
  const result = createHashWordArray(wordArray, numBytes);
  for (let src = firstByte, dst = 0; src < bytes.length; ++src, ++dst) {
    setByte(result, dst, bytes[src]);
  }
  return result;
};
```
It's performance is similar to what we currently have, the only noticeable advantage is that we don't join byte array of big integer to build a string (this operation might be implemented efficient by java script engine per se).
I did manual profiling of authentication session and this change does not have impact on runtime on my machine.

We can do better and directly build HashWordArray from big int internal representation as it does toByteArray() function. This code would be even more tricky, so I gave up. You still can give it a try :)

To construct BigInteger from HashWordArray we could also try to follow the code of fromHex() constructor. I also gave up on this :)
